### PR TITLE
Add support for `GET /auth`

### PR DESCRIPTION
**Pull request by stephank**
Required as per OAuth2: https://tools.ietf.org/html/rfc6749#section-3.1

See discussion in #13

### CODE REVIEW

File: src/lib.rs.in [hunk 1]
```diff
@@ -136,9 +137,14 @@ pub struct AuthHandler { pub app: AppConfig }
 impl Handler for AuthHandler {
     fn handle(&self, req: &mut Request) -> IronResult<Response> {
         info!("{} {}", req.method, req.url);
-        let params = try!(req.get_ref::<UrlEncodedBody>().map_err(
-                          |e| IronError::new(e, (status::BadRequest,
-                                                 "no query string in POST data"))));
+        let params = try!(
+            match req.method {
+                Method::Get => req.get_ref::<UrlEncodedQuery>(),
+                _ => req.get_ref::<UrlEncodedBody>()
```
Comment (djc) [inline src/lib.rs.in]: This should probably be explicitly for `Method::Post`, then the `_` case can raise an `Error`.

File: src/lib.rs.in [hunk 2]
```diff
@@ -136,9 +137,14 @@ pub struct AuthHandler { pub app: AppConfig }
...
+            }
+                .map_err(|e| IronError::new(e, (status::BadRequest,
+                                                "no query string in POST data")))
```
Comment (djc) [inline src/lib.rs.in]: The error string here is not correct anymore, please update it.

Comment (stephank) [inline src/lib.rs.in]: Whoops! Fixed.

File: src/lib.rs.in (hunk 1 above)
Comment (stephank) [inline src/lib.rs.in]: I agree, but couldn't think a way out of this one when I wrote it. Now it seems obvious. :)

**Comment by djc** [comment]
Sorry, forgot about this. Looking good, thanks!

**Pull request merged**